packages/babel-plugin-proposal-class-properties/src/index.js

```diff
@@ -219,22 +239,52 @@ export default function(api, options) {
           }
         }
 
+        let iife;
+
+        if (staticNodes.length > 0) {
+          // prepare iife before removing properties
+          iife = getClassIIFE(path, assumePure);
+        }
+
         for (const prop of props) {
           prop.remove();
         }
 
         if (computedNodes.length === 0 && afterNodes.length === 0) return;
 
-        if (path.isClassExpression()) {
-          path.scope.push({ id: ref });
-          path.replaceWith(t.assignmentExpression("=", ref, path.node));
+        if (isClassExpression) {
+          const outerRef = path.scope.generateUidIdentifier(ref.name);
+          path.scope.push({ id: outerRef });
+          path.replaceWith(t.assignmentExpression("=", outerRef, path.node));
```

Comment:
Q: What is outer ref needed for? It doesn't conflict now so it works, but I still don't understand what it is for.

Reply:
Im pretty confident it's not needed in 100% cases, it was here before the PR - didnt want to optimize the output more than needed. Gonna revisit this in followup PR.

Is this the only part that you do not understand? Or is something else needing a clarification too?

Reply:
I think it was needed because static properties where defined after the iife: https://babeljs.io/repl/build/master#?babili=false&browsers=&build=&builtIns=false&code_lz=BQYwNghgzlAECCsDeAoWsoBcKYJYlgDMB7Y2AXlgCYBuNWAWwFNMALYgE2AEpkBfFH240gA&debug=false&circleciRepo=&evaluate=false&fileSize=false&lineWrap=false&presets=es2015%2Cstage-0%2Ctypescript&prettier=false&targets=&version=7.0.0-beta.36 (`_class` is used in `Object.defineProperty`)

> Is this the only part that you do not understand? Or is something else needing a clarification too?

Thank you, the other parts are clear enough :blush: